src/Api/LockFreeApiEventProducer.cpp

```diff
@@ -0,0 +1,122 @@
+                        },
+                        [capture_event](const ApiTrackDouble& track_double) {
+                          auto* api_event = capture_event->mutable_api_track_double();
+                          SetMetaData(track_double.meta_data, api_event);
+                          SetEncodedName(track_double.encoded_name, api_event);
+                          api_event->set_data(track_double.data);
+                          api_event->set_color_rgba(track_double.color_rgba);
+                        },
+                        [capture_event](const ApiTrackFloat& track_float) {
+                          auto* api_event = capture_event->mutable_api_track_float();
+                          SetMetaData(track_float.meta_data, api_event);
+                          SetEncodedName(track_float.encoded_name, api_event);
+                          api_event->set_data(track_float.data);
+                          api_event->set_color_rgba(track_float.color_rgba);
+                        },
+                        [capture_event](const ApiTrackInt& track_int) {
+                          auto* api_event = capture_event->mutable_api_track_int();
+                          SetMetaData(track_int.meta_data, api_event);
+                          SetEncodedName(track_int.encoded_name, api_event);
+                          api_event->set_data(track_int.data);
+                          api_event->set_color_rgba(track_int.color_rgba);
+                        },
+                        [capture_event](const ApiTrackInt64& track_int64) {
+                          auto* api_event = capture_event->mutable_api_track_int64();
+                          SetMetaData(track_int64.meta_data, api_event);
+                          SetEncodedName(track_int64.encoded_name, api_event);
+                          api_event->set_data(track_int64.data);
+                          api_event->set_color_rgba(track_int64.color_rgba);
+                        },
+                        [capture_event](const ApiTrackUint& track_uint) {
+                          auto* api_event = capture_event->mutable_api_track_uint();
+                          SetMetaData(track_uint.meta_data, api_event);
+                          SetEncodedName(track_uint.encoded_name, api_event);
+                          api_event->set_data(track_uint.data);
+                          api_event->set_color_rgba(track_uint.color_rgba);
+                        },
+                        [capture_event](const ApiTrackUint64& track_uint64) {
+                          auto* api_event = capture_event->mutable_api_track_uint64();
+                          SetMetaData(track_uint64.meta_data, api_event);
+                          SetEncodedName(track_uint64.encoded_name, api_event);
+                          api_event->set_data(track_uint64.data);
+                          api_event->set_color_rgba(track_uint64.color_rgba);
+                        }},
+             raw_api_event);
```

Comment:
It is rather a question. looks like this is a alternative to producing a separate override for each variant right?

This is for sure more compact I wonder if it is still readable.

Reply:
See above. As you both were unsure about readability, I went with a set of overloaded functions.